broker-core/src/main/java/io/zeebe/broker/exporter/context/ExporterConfiguration.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Zeebe Broker Core
+ * Copyright © 2017 camunda services GmbH (info@camunda.com)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package io.zeebe.broker.exporter.context;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonElement;
+import io.zeebe.exporter.context.Configuration;
+import java.util.Map;
+
+public class ExporterConfiguration implements Configuration {
+  private static final Gson CONFIG_INSTANTIATOR = new GsonBuilder().create();
+
```

Comment:
The TOML parser we use doesn't have a way to deserialize a Map to an Object; I'd need to reserialize it to TOML then deserialize it again.

I used `Gson` because that's what the TOML parser uses under the covers, and I wanted to remain as close as possible. See: https://github.com/mwanji/toml4j/blob/4d206e4af6f0dca7643fcfb043235a970d99f810/src/main/java/com/moandjiezana/toml/Toml.java#L316

The ideal solution would to somehow get a partial TOML document for just that portion, but I couldn't figure out how to do this in a nice way.

Reply:
But why do you want that? Is there any use case for this? Would

Reply:
Right now the exporter configuration is a nested map, and we pass that map to the exporter so it can configure itself. The Gson part is used for the facility we provide to the user to unmarshal that map into a class of the user's choice. It's not required, since we also expose the map object itself, so they can perform the deserialization themselves, but I think it's useful. I can switch to Jackson and it will probably work, but I figured let's be safe and do it the same way our TOML parser does.

Reply:
My point is do we really need that? :sweat_smile: But yeah anyway since it is not real complex we can keep it as it is